### ISSUE TITLE
where is the source code for the Vanced app

### ISSUE THREAD

**Issue by daubneyi** [YTVanced/Vanced#17]
I see source for Vanced Manager and MicroG, but why is there not source available for the Vanced YouTube app?

**Comment by KevinX8** [comment]
The app is not built from source, how do you propose we create it?

**Comment by daubneyi** [comment]
👌🏻😩

**Comment by varuzam** [comment]
is there any instruction how to build vanced app from original youtube app? any scripts etc..

**Comment by X1nto** [comment]
> is there any instruction how to build vanced app from original youtube app? any scripts etc..

nope

**Comment by varuzam** [comment]
do they not exist or you dont want to open them? I am not an expert at repacking apks. but I cannot imagine how it can be achieved without any scripts/code/docs.

**Comment by KevinX8** [comment]
All the modding can be done manually wdym, we have patcher to help automate it but that will never be open sourced

**Comment by Diegovsky** [comment]
Could you please release it? Privacy is a big concern for me so I won't be able to use vanced if I don't trust it

**Comment by milindgoel15** [comment]
If you don't trust it, don't use it. No one is forcing you to install and use. Vanced is closed source as official yt is in the first place.

**Comment by Diegovsky** [comment]
Thanks for the kind response. Yes, the original YT app is closed source; that's precisely why I don't want to use it. I know this is a difference in opinion, but I don't see a reason why Vanced must be closed source in the first place. There isn't a "competition" to hide company secrets from, unless you're doing something you can't tell the user. Your attitude and indifference make me believe that is the case.

What am I going to do then? Nothing; we are adults after all, aren't we? Thanks for your time.

**Comment by X1nto** [comment]
> Thanks for the kind response. Yes, the original YT app is closed source; that's precisely why I don't want to use it. I know this is a difference in opinion, but I don't see a reason why Vanced must be closed source in the first place. There isn't a "competition" to hide company secrets from, unless you're doing something you can't tell the user. Your attitude and indifference make me believe that is the case.
> 
> What am I going to do then? Nothing; we are adults after all, aren't we? Thanks for your time.

We're trying to say that Vanced can't possibly be open source as stock YouTube itself is closed, which makes it impossible for Vanced to be open-sourced. It's all just bytecode modifications inside an already obfuscated YouTube dex.

**Comment by Diegovsky** [comment]
Thanks for the explanation. I understand it better now. I wasn't being far either; let me rephrase my question: Is it possible to open source the patches and processes used to create the vanced apk?

**Comment by X1nto** [comment]
> Thanks for the explanation. I understand it better now. I wasn't being far either; let me rephrase my question: Is it possible to open source the patches and processes used to create the vanced apk?

Oh, sadly no. If we open up the patcher, google will able to find what Vanced does and obfuscate their code, even more. This will prevent us from developing future Vanced versions.

**Comment by peepopoggers** [comment]
> > Thanks for the explanation. I understand it better now. I wasn't being far either; let me rephrase my question: Is it possible to open source the patches and processes used to create the vanced apk?
> 
> Oh, sadly no. If we open up the patcher, google will able to find what Vanced does and obfuscate their code, even more. This will prevent us from developing future Vanced versions.

Surely then you could only make the grabber/extractor parts of the app proprietary? This seems unmanaged.

**Comment by KevinX8** [comment]
> > > Thanks for the explanation. I understand it better now. I wasn't being far either; let me rephrase my question: Is it possible to open source the patches and processes used to create the vanced apk?
> > 
> > 
> > Oh, sadly no. If we open up the patcher, google will able to find what Vanced does and obfuscate their code, even more. This will prevent us from developing future Vanced versions.
> 
> Surely then you could only make the grabber/extractor parts of the app proprietary? This seems unmanaged.

The other parts don't have source code, they're literally just smali mods which you can get by decompiling, the only mod with source code is the sponsorblock part and it is open source.

**Comment by peepopoggers** [comment]
ok

**Comment by VanceAlot** [comment]
> is there any instruction how to build vanced app from original youtube app? any scripts etc..

++ did you find any? I can't even find the original patches.

**Comment by Daasin** [comment]
Or just the added code, only release the code that is added in/modified
